src/paperoni/__main__.py

```diff
@@ -199,14 +199,20 @@ class Get(Productor):
 
         def run(self, work: "Work"):
             ex = work.collection and work.collection.exclusions
-            mem_col = MemCollection(
-                _last_id=work.collection._last_id if work.collection else None,
+
+            find = Finder(
+                title_finder=lambda scored: scored.value.current.title,
+                links_finder=lambda scored: scored.value.current.links,
+                id_finder=lambda scored: getattr(scored.value.current, "id", None),
             )
-            mem_col.add_papers(scored.value.current for scored in work.top)
+            find.add(list(work.top))
+
             for pinfo in self.iterate(focuses=work.focuses):
                 if ex and pinfo.key in ex:
                     continue
-                if mem_col.find_paper(pinfo.paper):
+
+                if found := find.find(pinfo.paper):
+                    found.value.add(pinfo)
```

Comment:
Is scored automatically updated here or is there another operation that needs to be done? Something like the following?


```suggestion
                    found.value.add(pinfo)
                    found.score = work.focuses.score(found.value.current)
```